models/path.py

```diff
@@ -32,16 +33,20 @@ def link_affected_by_interface(self, interface=None):
                 return link
         return None
 
-    def choose_vlans(self):
+    def choose_vlans(self, controller):
         """Choose the VLANs to be used for the circuit."""
         for link in self:
-            tag = link.get_next_available_tag()
+            tag_value = link.get_next_available_tag(controller)
+            tag = TAG('vlan', tag_value)
             link.add_metadata("s_vlan", tag)
 
-    def make_vlans_available(self):
+    def make_vlans_available(self, controller):
         """Make the VLANs used in a path available when undeployed."""
         for link in self:
-            link.make_tag_available(link.get_metadata("s_vlan"))
+            tag = link.get_metadata("s_vlan")
+            link.make_tag_available(
```

Comment:
This is crashing when deleting an EVC:

```
    link.make_tag_available(
AttributeError: 'Link' object has no attribute 'make_tag_available'
```

Reply:
This is a type it is supposed to be `..._tags_...` (plural), somehow I changed it back.